packages/contracts-bedrock/src/L1/StandardValidator.sol

```diff
@@ -0,0 +1,271 @@
+    string internal constant preimageOracleVersion = "1.1.2";
+
+    /// Bytecode hashes of our contracts. Does not include contracts that contain immutables. Pulled from
+    /// the Superchain Registry.
+
+    bytes32 internal constant systemConfigHash = 0x165dece6291f3a071e602c804f4a3e89977a71ec866f46c85eb21448f22f6f9a;
+    bytes32 internal constant permissionedDisputeGameHash =
+        0x1a81c5fe9fbbf68a0f75e987451e5181c4fc3e801f5c056d38f08942cefed349;
+    bytes32 internal constant mipsHash = 0xd80e53cfb3a0655e5a82b40f370b7352bd926d6d28de7f1735583fcfeba3f0fd;
+    bytes32 internal constant optimismPortalHash = 0x2b0610230634fccf9d05f4aaefceffef9e04aa99910798d42823e0068eb07eca;
+    bytes32 internal constant anchorStateRegistryHash =
+        0x165dece6291f3a071e602c804f4a3e89977a71ec866f46c85eb21448f22f6f9a;
+    bytes32 internal constant delayedWETHHash = 0x1a81c5fe9fbbf68a0f75e987451e5181c4fc3e801f5c056d38f08942cefed349;
+    bytes32 internal constant disputeGameFactoryHash =
+        0xd80e53cfb3a0655e5a82b40f370b7352bd926d6d28de7f1735583fcfeba3f0fd;
+    bytes32 internal constant l1CrossDomainMessengerHash =
+        0xc919b535f7bbbaa57f32b602f26000e3997c22c933efbb92a36e053b20b2b4b1;
+    bytes32 internal constant l1ERC721BridgeHash = 0xa1a3ea3996797cf5035dafff9335386f8b00c58ff22aaa4c34b02c4ae7cf5e8a;
+    bytes32 internal constant l1StandardBridgeHash = 0x6894839591ba2aa1f6ba43526113e25fd795ec5c5a48202e664c931cd170aaee;
+    bytes32 internal constant optimismMintableERC20FactoryHash =
+        0xb8ec72533005496795f8fc0c71f0312578ff626f7b94699984e1add6dc2f3b88;
+    bytes32 internal constant preimageOracleHash = 0x2109eb20fb34703fe7ab0231bf0e93d972fe78423b72e7fb1f797f168217b87d;
+
+    address internal constant defaultL2Sender = 0x000000000000000000000000000000000000dEaD;
+
+    bool internal isFailed = false;
+
+    string internal collectedErrors = "";
+
+    struct Input {
+        IProxyAdmin proxyAdmin;
+        ISystemConfig sysCfg;
+        bytes32 absolutePrestate;
+        uint256 l2ChainID;
+    }
+
+    constructor(ISuperchainConfig _superchainConfig, address _l1PAOMultisig, address _mips) {
+        superchainConfig = _superchainConfig;
+        l1PAOMultisig = _l1PAOMultisig;
+        mips = _mips;
+    }
+
+    function validate(StandardValidatorV180.Input memory _input, bool _allowFailure) public returns (string memory) {
+        assertValidProxyAdmin(_input.proxyAdmin);
+        address _l1PAO = _input.proxyAdmin.owner();
+
+        ISystemConfig _sysCfg = _input.sysCfg;
+        assertValidSystemConfig(_sysCfg);
+        assertValidL1CrossDomainMessenger(_sysCfg);
+        assertValidOptimismMintableERC20Factory(_sysCfg);
+        assertValidL1ERC721Bridge(_sysCfg);
+        assertValidOptimismPortal(_sysCfg);
+        assertValidDisputeGameFactory(_sysCfg, _l1PAO);
+        assertValidPermissionedDisputeGame(_sysCfg, _l1PAO, _input.absolutePrestate, _input.l2ChainID);
```

Comment:
we should have L1 standard bridge, anchor state registry, and permissionless dispute game validator methods

Reply:
we should have a superchain config validator method too, to make sure e.g. the guardian is correct and the chain is unpaused (I can't imagine why we would want to run this if the chain is paused)

Reply:
ASR is validated as part of the dispute game, since you have to get it from the dispute game itself.